platforms/core-runtime/launcher/src/integTest/groovy/org/gradle/launcher/daemon/SingleUseDaemonIntegrationTest.groovy

```diff
@@ -117,6 +138,24 @@ assert java.lang.management.ManagementFactory.runtimeMXBean.inputArguments.conta
         wasNotForked()
     }
 
+    @Requires(IntegTestPreconditions.JavaHomeWithDifferentVersionAvailable)
+    def "does not fork build when daemon jvm criteria from build properties matches current process"() {
+        def otherJdk = AvailableJavaHomes.differentVersion
```

Comment:
Instead of testing against _another_ java home, can we just test against `Jvm.current()`? Then, we wouldn't need to override the executer's java home

This ensures the java versions on CI correspond with the java versions tested in the test

Reply:
This test represents a copy of the `forks build with default daemon JVM args when java home from gradle properties does not match current process` but applied for the daemon toolchain. It definitely makes sense to test against different JDK than the `JAVA_HOME` since the `Jvm.current()` on the build execution is expected to be different than the `JAVA_HOME`. 

In addition, this test is validating that daemon runs in a new process since doesn't match with the existing Launcher JVM that is using the JAVA_HOME